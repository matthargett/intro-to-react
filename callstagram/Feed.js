import React from "react";
import FeedItem from "./FeedItem";

import { connect } from "react-redux";

import { ScrollView } from "react-native";

const FeedWithItems = props => {
  return (
    <ScrollView>
      {props.itemIDs.map(id => <FeedItem id={id} key={id} />)}
    </ScrollView>
  );
};

const Feed = connect(store => {
  return {
    itemIDs: store.items.map(i => i.id)
  };
})(FeedWithItems);

export default Feed;
