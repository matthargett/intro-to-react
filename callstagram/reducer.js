const initialState = {
  items: [],
  likedItems: [],
  layout: {}
};

const reducer = (lastState = initialState, action) => {
  if (action.type === "DOGGIE_DELIVERY") {
    return {
      ...lastState,
      items: [...lastState.items, ...action.images]
    };
  }
  if (action.type === "LAYOUT") {
    return {
      ...lastState,
      layout: action.layout
    };
  }
  if (action.type === "TOGGLE_LIKE") {
    const hasLiked = lastState.likedItems.indexOf(action.id) !== -1;
    if (hasLiked) {
      return {
        ...lastState,
        likedItems: lastState.likedItems.filter(i => i !== action.id)
      };
    } else {
      return {
        ...lastState,
        likedItems: [...lastState.likedItems, action.id]
      };
    }
  }
  return lastState;
};

export default reducer;
