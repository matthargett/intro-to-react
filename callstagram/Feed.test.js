import React from "react";
import Feed from "./Feed";
import reducer from "./reducer";

import { createStore } from "redux";
import { Provider } from "react-redux";

import renderer from "react-test-renderer";

it("renders loading state", () => {
  const store = createStore(reducer);
  const rendered = renderer
    .create(
      <Provider store={store}>
        <Feed />
      </Provider>
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

it("renders with delivered images", () => {
  const store = createStore(reducer);
  store.dispatch({
    type: "DOGGIE_DELIVERY",
    images: [{ key: "asdf", image: "http://imageURL" }]
  });
  const rendered = renderer
    .create(
      <Provider store={store}>
        <Feed store={store} />
      </Provider>
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});
