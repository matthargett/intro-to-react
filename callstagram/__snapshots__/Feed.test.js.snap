// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading state 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    Loading..
  </Text>
</View>
`;

exports[`renders with delivered images 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "image": "http://imageURL",
        "key": "asdf",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={3}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={2}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={Object {}}
        >
          <Image
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={
              Object {
                "uri": "http://imageURL",
              }
            }
            style={
              Object {
                "aspectRatio": 1,
                "width": NaN,
              }
            }
            testID={undefined}
          />
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={undefined}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Like Me
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
